js/coinone.js

```diff
@@ -371,78 +371,253 @@ module.exports = class coinone extends Exchange {
+            if (side.indexOf ('ask') >= 0) {
+                side = 'sell';
+                feeCurrency = market['quote'];
+            } else {
+                side = 'buy';
+                feeCurrency = market['base'];
+            }
+            const amount = this.safeFloat (order, 'qty');
+            result.push ({
+                'id': this.safeString (order, 'orderId'),
+                'clientOrderId': undefined,
+                'timestamp': timestamp,
+                'datetime': this.iso8601 (timestamp),
+                'lastTradeTimestamp': undefined,
+                'status': 'open',
+                'symbol': market['symbol'],
+                'type': 'limit',
+                'side': side,
+                'price': this.safeFloat (order, 'price'),
+                'amount': amount,
+                'filled': 0,            // coinone set amount as same as remaining, different with ordered amount
+                'remaining': amount,    // coinone set amount as same as remaining, different with ordered amount
+                'cost': 0,              // coinone set amount as same as remaining, different with ordered amount
+                'trades': undefined,    // coinone doesn't support fills
+                'fee': {
+                    'currency': feeCurrency,
+                    'rate': this.safeFloat (order, 'feeRate'),
+                    'cost': 0,          // coinone set amount as same as remaining, different with ordered amount
+                },
+                'info': order,
+            });
+        }
+        return result;
+    }
+
+    async fetchClosedOrders (symbol = undefined, since = undefined, limit = undefined, params = {}) {
+        // [WARNING] Because of the limited capabilities of Coinone V2 API, you should be noticed below
+        // * The symbol parameter must be given. If not, an exception will occur.
+        // * The since and limit parameters are totally ignored.
+        // * At maximum, 50 closed orders can be returned.
+        // * An returned closed order might be in open status: Because the backing Coinone API returns my trades, a parially-filled but-not-closed order can be returned.
+        // * For the same reason, a same order id can occur muliple times in the returned list.
+        // * For the same reason, the returned amount and filled are always same, and the returned remaining is always zero.
+        await this.loadMarkets ();
+        if (symbol === undefined) {
+            throw new ExchangeError (this.id + ' allows fetching closed orders with a specific symbol');
+        }
+        const market = this.market (symbol);
+        const request = {
+            'currency': market['id'],
+        };
+        const response = await this.privatePostOrderCompleteOrders (this.extend (request, params));
+        const result = [];
+        const completOrders = this.safeValue (response, 'completeOrders');
+        for (let i = 0; i < completOrders.length; i++) {
```

Comment:
We need to move this parsing section to `parseOrder()`, to keep it in line with almost all other exchanges.

Reply:
The semantic of 'order' from Coinone privatePostOrderCompleteOrders() and privatePostOrderLimitOrders() are not same.
So we cannot write unified parsing logic for them, I think.